shlyapa/shlyapa.py

```diff
@@ -0,0 +1,171 @@
+import next_pair_alg
+from config import *
+from pair import *
+import random
+
+
+class Game:
+    class Explanation:
+        def __init__(self, pair=Pair(), number_explained=0):
+            self.pair = Pair(pair.explaining, pair.guessing)
+            self.number_explained = number_explained
+
+    class Tour:
+        def __init__(self, number_words):
+            self.explanations = []
+            self.number_words = number_words
+            self.number_explained_words = 0
+
+        def add_explanation(self, explanation):
+            if self.is_end():
+                raise ValueError("error add to ended tour")
+            if self.number_explained_words + explanation.number_explained > self.number_words:
+                raise ValueError("too many words in tour")
+
+            self.number_explained_words += explanation.number_explained
+            self.explanations.append(explanation)
+
+        def pop_explanation(self):
+            if self.is_new():
+                ValueError("error pop_explanation from new tour")
+
+            assert (self.number_explained_words >= self.explanations[-1].number_explained)
+
+            self.number_explained_words -= self.explanations[-1].number_explained
+            self.explanations.pop()
+
+        def is_new(self):
+            return len(self.explanations) == 0
+
+        def is_end(self):
+            return self.number_explained_words == self.number_words
+
+    def __init__(self, config_filepath=default_config_game_filepath):
+        self.config = Config(config_filepath)
+        self.__tours = []
+        self.__cur_turn = 0
+        # First pair in sequence of pairs who have not play yet
+        self.__next_pair = Pair()
+        self.__number_explained_words = 0
+        self.__alg = None
+        if self.config.type == Config.TypeGame.ORIGINAL:
```

Comment:
In face TypeGame is redundant - you can use classes as IDs

here should be just simple instantiation:
```suggestion
        self.__alg = self.config.type()
```

see also comment to config class